File: app/src/main/java/com/mtkdownload/GeneralRunnable.java
package com.mtkdownload;

import android.os.Bundle;
import android.os.Handler;
import android.os.Message;

/** Nested class that performs the restart */
public class GeneralRunnable implements Runnable 
{
	Handler mHandler;
	protected static String GPS_bluetooth_id;

	GeneralRunnable(Handler h) 
	{
		mHandler = h;
		GPS_bluetooth_id = MTKDownload.getSharedPreferences().getString("bluetoothListPref", "-1");
	}

	protected void sendTOAST(String message) {
		Message msg = mHandler.obtainMessage();
		Bundle b = new Bundle();
		b.putString(MTKDownload.KEY_TOAST, message);
		msg.setData(b);
		mHandler.sendMessage(msg);
	}

	protected void sendMessageField(String message) {
		Message msg = mHandler.obtainMessage();
		Bundle b = new Bundle();
		b.putString(MTKDownload.MESSAGEFIELD, message);
		msg.setData(b);
		mHandler.sendMessage(msg);
	}
	
	protected void sendSettings_MessageField(String message) {
		Message msg = mHandler.obtainMessage();
		Bundle b = new Bundle();
		b.putString(MTKDownload.SETTINGS_MESSAGEFIELD, message);
		msg.setData(b);
		mHandler.sendMessage(msg);
	}

	protected void sendCloseProgress() {
		Message msg = mHandler.obtainMessage();
		Bundle b = new Bundle();
		b.putInt(MTKDownload.CLOSE_PROGRESS, 1);
		msg.setData(b);
		mHandler.sendMessage(msg);
	}

	@Override
	public void run() {
		
	}

}

File: app/src/main/java/com/mtkdownload/GeneralRunnable.kt
package com.mtkdownload

import android.os.Bundle
import android.os.Handler

/** Nested class that performs the restart  */
open class GeneralRunnable internal constructor(val mtkDownload: MTKDownload,var mHandler: Handler) : Runnable {
    init {
        GPS_bluetooth_id = mtkDownload.sharedPreferences.getString("bluetoothListPref", "-1").toString()
    }

    protected fun sendTOAST(message: String?) {
        val msg = mHandler.obtainMessage()
        val b = Bundle()
        b.putString(mtkDownload.KEY_TOAST, message)
        msg.data = b
        mHandler.sendMessage(msg)
    }

    protected fun sendMessageField(message: String?) {
        val msg = mHandler.obtainMessage()
        val b = Bundle()
        b.putString(mtkDownload.MESSAGEFIELD, message)
        msg.data = b
        mHandler.sendMessage(msg)
    }

    protected fun sendSettings_MessageField(message: String?) {
        val msg = mHandler.obtainMessage()
        val b = Bundle()
        b.putString(mtkDownload.SETTINGS_MESSAGEFIELD, message)
        msg.data = b
        mHandler.sendMessage(msg)
    }

    protected fun sendCloseProgress() {
        val msg = mHandler.obtainMessage()
        val b = Bundle()
        b.putInt(mtkDownload.CLOSE_PROGRESS, 1)
        msg.data = b
        mHandler.sendMessage(msg)
    }

    override fun run() {}

    companion object {
        protected var GPS_bluetooth_id: String=""
    }
}